infinispan-remote/src/test/java/org/hibernate/ogm/datastore/infinispanremote/test/sequences/SequenceAtomicIncrementTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.infinispanremote.test.sequences;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.lucene.util.FixedBitSet;
+import org.hibernate.ogm.backendtck.id.Actor;
+import org.hibernate.ogm.cfg.OgmProperties;
+import org.hibernate.ogm.datastore.infinispanremote.InfinispanRemoteProperties;
+import org.hibernate.ogm.datastore.infinispanremote.impl.InfinispanRemoteDatastoreProvider;
+import org.hibernate.ogm.datastore.infinispanremote.impl.sequences.HotRodSequenceHandler;
+import org.hibernate.ogm.datastore.infinispanremote.utils.InfinispanRemoteTestHelper;
+import org.hibernate.ogm.datastore.infinispanremote.utils.RemoteHotRodServerRule;
+import org.hibernate.ogm.dialect.spi.NextValueRequest;
+import org.hibernate.ogm.model.impl.DefaultIdSourceKeyMetadata;
+import org.hibernate.ogm.model.key.spi.IdSourceKey;
+import org.hibernate.ogm.utils.OgmSessionFactoryRule;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+/**
+ * Stress test to verify the Sequence generation is generating unique numbers even
+ * when under load by parallel requests.
+ * If this test passes it does not guarantee that the implementation is correct;
+ * if it fails there certainly is a problem.
+ *
+ * By default it's not running much of a stress test as we want the tests to give
+ * quick feedback; Set the ogm.runFullStressTests System property to have it run
+ * for several minutes.
+ * Careful when enlarging constants for longer tests to also allow for more memory
+ * as we need to track and compare all generated results at the end.
+ *
+ * @author Sanne Grinovero
+ */
+public class SequenceAtomicIncrementTest {
```

Comment:
I'm not sure it's worth it. The semantics have been carefully inspected by people on the Infinispan team, if there are failures *they* should detect them with their tests.
Also it's not meant to be a long term solution; we're discussing better alternative either using a custom task which we can deploy, or to have Hot Rod expose a sequence functionality.